### PR TITLE
fix: keep static properties when using @external decorator

### PR DESCRIPTION
**Pull request by misantronic**
at the moment, static properties will be lost when using the @external decorator.
this fix iterates them and copies the static properties to the constructor.

### CODE REVIEW

**Comment by KnisterPeter** [comment]
Please add test to your fix

File: lib/decorators.ts [hunk 1]
```diff
@@ -371,7 +371,10 @@ export function External<TFunction extends Function>(...args: any[]): ClassDecor
       return (target as any).__tsdi__.configureExternal(args, target);
     };
     (constructor as any).displayName = (target as any).name;
-    constructor.prototype = target.prototype;
+
+    Object.getOwnPropertyNames(target)
+      .filter(prop => prop !== '__tsdi__' && prop !== 'name' && prop !== 'length')
```
Comment (KnisterPeter) [inline lib/decorators.ts]: why do you filter `__tsdi__` and `length`?

Comment (misantronic) [inline lib/decorators.ts]: it was originally not included so I thought I might ditch it.
If it doesn't matter, we could include it anyway.

Comment (KnisterPeter) [inline lib/decorators.ts]: Just add everything which does exist would be better I think.

Comment (misantronic) [inline lib/decorators.ts]: ok

**Pull request merged**